src/components/tooltip/tooltip.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { FC, HTMLAttributes, useState, ReactNode } from 'react'
+import { ElToolTipChild, ElToolTipContainer, ElToolTipLabel } from './styles'
+import { useId } from '#src/storybook/random-id'
+
+export interface ToolTipProps extends HTMLAttributes<HTMLDivElement> {
+  children: ReactNode
+  label?: string
+  tip: string
+  maxWidth?: string
+  position?:
+    | 'top'
+    | 'bottom'
+    | 'right'
+    | 'left'
+    | 'topStart'
+    | 'topEnd'
+    | 'bottomStart'
+    | 'bottomEnd'
+    | 'rightStart'
+    | 'rightEnd'
+    | 'leftStart'
+    | 'leftEnd'
+}
+
+export interface ToolTipChildProps extends HTMLAttributes<HTMLDivElement> {
+  position: ToolTipProps['position']
+  maxWidth?: string
+}
+
+export const ToolTip: FC<ToolTipProps> = ({ children, label, tip, maxWidth = '400px', position = 'top' }) => {
+  const [visible, setVisible] = useState(false)
+  const tooltipId = useId()
+
+  return (
+    <ElToolTipContainer
+      onMouseEnter={() => setVisible(true)}
+      onMouseLeave={() => setVisible(false)}
+      onFocus={() => setVisible(true)}
+      onBlur={() => setVisible(false)}
+      aria-describedby={tooltipId}
```

Comment:
**issue:** I'm not sure placing `aria-describedby` on the parent is correct. In the case where the `children` is a `<button>`, I believe the `aria-describedby` should be on the `<button>` element itself, not it's parent 🤔 

I suspect we need something a little more akin to `Menu.Trigger`